packages/store/src/StoreCore.sol

```diff
@@ -272,14 +278,20 @@ library StoreCore {
     // Emit event to notify indexers
     emit StoreSetField(tableId, key, schemaIndex, fullData);
 
-    // Call onSetField hooks (before actually modifying the state, so observers have access to the previous state if needed)
+    // Call preSetField hooks (before modifying the state)
     address[] memory hooks = Hooks.get(bytes32(tableId));
-    for (uint256 i = 0; i < hooks.length; i++) {
+    for (uint256 i; i < hooks.length; i++) {
       IStoreHook hook = IStoreHook(hooks[i]);
-      hook.onSetField(tableId, key, schemaIndex, fullData);
+      hook.preSetField(tableId, key, schemaIndex, fullData);
     }
 
     StoreCoreInternal._pushToDynamicField(tableId, key, schema, schemaIndex, dataToPush);
+
+    // Call postSetField hooks (after modifying the state)
+    for (uint256 i; i < hooks.length; i++) {
+      IStoreHook hook = IStoreHook(hooks[i]);
+      hook.postSetField(tableId, key, schemaIndex, fullData);
+    }
```

Comment:
not blocking, but thinking aloud: in terms of gas, I wonder if we could do anything to detect if a hook has implemented a given method and skip if not, or if adding a hook should be for a particular operation (e.g. `IStoreOnBeforeSetFieldHook`)

Reply:
Agree it's kind of unnecessary to call the hook method if it's not implemented but I'm afraid checking if the hook implements a certain interface would be more gas expensive. We can make a note of this for the gas optimization pass though! Will add it to #444.